spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/sampler/PercentageBasedSampler.java

```diff
@@ -0,0 +1,47 @@
+package org.springframework.cloud.sleuth.sampler;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cloud.sleuth.Sampler;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.TraceAccessor;
+
+/**
+ * Sampler that based on the given percentage rate will allow sampling.
+ *
+ * A couple of assumptions have to take place in order for the algorithm to work properly:
+ *
+ * <ul>
+ *     <li>We're taking the TraceID into consideration for sampling to be consistent</li>
+ *     <li>We treat the TraceID as a UUID - a 128 bit number - like HTrace does - https://github.com/apache/incubator-htrace/blob/master/htrace-core4/src/main/java/org/apache/htrace/core/SpanId.java</li>
+ *     <li>We apply the Finagle algorithm to define whether we should sample or not (we're doing a % 100 not greater) - https://github.com/openzipkin/zipkin-java/pull/55/files#diff-31b830705fabb2e936c73337dfa5f219R83</li>
```

Comment:
Change reference to zipkin algorithm https://github.com/openzipkin/zipkin-java/blob/master/zipkin-java-core/src/main/java/io/zipkin/TraceIdSampler.java#L78

(as this isn't the finagle one anymore)